src/frontend/components/PublicPausedLiveVideo/index.tsx

```diff
@@ -0,0 +1,95 @@
+import { Box, Clock, Layer, Paragraph } from 'grommet';
+import { DateTime } from 'luxon';
+import React, { useEffect, useState } from 'react';
+import { defineMessages, FormattedMessage } from 'react-intl';
+import videojs from 'video.js';
+
+import { H2 } from '../Headings';
+import { Video } from '../../types/tracks';
+import { useAsyncEffect } from 'utils/useAsyncEffect';
+import { resumeLive } from '../../utils/resumeLive';
+
+const messages = defineMessages({
+  pausedSince: {
+    defaultMessage: 'The webinar is in paused since ',
+    description: 'Information about how many time the webinar is paused.',
+    id: 'components.PublicPausedLive.pausedSince',
+  },
+  title: {
+    defaultMessage: 'Webinar is paused',
+    description: 'Title for the public paused live component',
+    id: 'components.PublicPausedLive.title',
+  },
+  text: {
+    defaultMessage:
+      'The webinar is paused. When resumed, the video will start again.',
+    description: 'Message for the public paused live component',
+    id: 'components.PublicPausedLive.message',
+  },
+});
+
+interface PublicPausedLiveVideoProps {
+  video: Video;
+  videoNodeRef: HTMLVideoElement;
+}
+
+export const PublicPausedLiveVideo = ({
+  video,
+  videoNodeRef,
+}: PublicPausedLiveVideoProps) => {
+  const [time, setTime] = useState<string>();
+
+  useEffect(() => {
```

Comment:
I tnink you can use a simple `useMemo` here instead of this `useState` and `useEffect` since you want ton compute this value only once
this will "prevent" a second render when the `setState` apply

Reply:
This code has evolved since your comment. I also have to display since how many days the video is in paused. The `useEffect` now is:

```
const [time, setTime] = useState<string>();
  const [days, setDays] = useState(0);

  useEffect(() => {
    if (video.live_info?.paused_at) {
      const now = DateTime.now();
      const pausedAt = DateTime.fromSeconds(Number(video.live_info?.paused_at));
      const diff = now
        .diff(pausedAt, ['days', 'hours', 'minutes', 'seconds'])
        .toObject();
      // Using diff.toISO() would be better but Clock grommet component is not able to manage
      // iso8601 period without hours.
      setDays(diff.days || 0);
      setTime(`PT${diff.hours}H${diff.minutes}M${diff.seconds}S`);
    }
  }, []);
```


Also, the second parameter of `useEffect` is an empty array, I thought that the `useEffect` will never been executed after when `state` or `props` are updated.

Reply:
This can do the trick:
```
  const diffSincePause = React.useMemo<DurationObjectUnits | null>(() => {
    if (video.live_info?.paused_at === undefined) {
      return null;
    }

    const now = DateTime.now();
    const pausedAt = DateTime.fromSeconds(Number(video.live_info?.paused_at));
    return now
      .diff(pausedAt, ['days', 'hours', 'minutes', 'seconds'])
      .toObject();
  }, []);
  const time = React.useMemo<string | undefined>(() => {
    if (!diffSincePause) {
      return undefined;
    }

    return `PT${diffSincePause.hours}H${diffSincePause.minutes}M${diffSincePause.seconds}S`;
  }, [diffSincePause]);
  const [days, setDays] = useState(diffSincePause?.days ?? 0);
```

You are right about the dependencies array for the useEffect, I just find it quitte overkill to use a useEffect to initialize a constant inside a component that doesn't need cleanup

Reply:
Nice :+1: 
Thanks